db/version_set.cc

```diff
@@ -3223,6 +3252,27 @@ bool VersionStorageInfo::RangeMightExistAfterSortedRun(
   return false;
 }
 
+void VersionStorageInfo::SampleFileReadAndCheckThreshold(
+    FileMetaData* file_meta, bool* file_above_read_threshold) {
+  assert(file_meta != nullptr);
+  sample_file_read_inc(file_meta);
```

Comment:
Once these functions are made inline, I guess it might be worth replacing the two other calls to `sample_file_read_inc` in this file (in iterator code path) with `SampleFileReadAndCheckThreshold` and a nullptr for `file_above_read_threshold`. That would help us reduce the code changes when we are ready to extend read-triggered-compactions feature to iterators in the future.